core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1010,6 +1010,67 @@ private[spark] object Utils extends Logging {
     )
   }
 
+  val timeSuffixes = Map (
+    "us" -> TimeUnit.MICROSECONDS,
+    "ms" -> TimeUnit.MILLISECONDS,
+    "s" -> TimeUnit.SECONDS,
+    "min" -> TimeUnit.MINUTES,
+    "h" -> TimeUnit.HOURS,
+    "d" -> TimeUnit.DAYS
+  )
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a microsecond count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+   def parseTimeString(str: String) : (Option[TimeUnit], Long) = {
```

Comment:
I see - you're relying on the order of the map to resolve conflicts. You could still have a one-liner for that but it would be a little less efficient, so the latest code is fine.